.github/workflows/tests.yml

```diff
@@ -13,19 +13,15 @@ concurrency:
 jobs:
   tests:
     if: github.repository_owner == 'Qiskit'
-    name: macOS-arm64-tests-Python-${{ matrix.python-version }}
-    runs-on: macOS-14
+    name: ${{ matrix.os }}-arm64-tests-Python-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
```

Comment:
We branch-protect on these names - we'll need to update them in the protection settings before this can merge.